lib/utils.coffee

```diff
@@ -292,10 +293,12 @@ timeoutPromise = (ms, promise) ->
 # utils.requestAsync({ url: 'http://example.com' }).then (response) ->
 # 	console.log(response)
 ###
-exports.requestAsync = (options) ->
+exports.requestAsync = (options, retriesRemaining = undefined) ->
```

Comment:
I personally really dislike it as it makes me think there is some special reason it needs to be `undefined` rather than just nullish, so I'm watching out for that in the code and it adds processing overhead from that.  I'd rather just have it in the jsdoc due to that, also for typescript I would write it as `retriesRemaining?: number` (or `retriesRemaining: number | null`, I'm not sure which I prefer yet) rather than `retriesRemaining: number | null = undefined`